src/main/java/slimeknights/tconstruct/plugin/jei/JEIPlugin.java

```diff
@@ -204,6 +209,15 @@ public void registerRecipes(IRecipeRegistration register) {
     List<SeveringRecipe> severingRecipes = RecipeHelper.getJEIRecipes(manager, TinkerRecipeTypes.SEVERING.get(), SeveringRecipe.class);
     register.addRecipes(TConstructJEIConstants.SEVERING, severingRecipes);
 
+    // tool recipes
+    List<ToolRecipes> toolRecipes = ToolDefinitionLoader.getInstance().getRegisteredToolDefinitions()
+      .stream()
+      .filter(definition -> definition.hasMaterials() && !definition.getId().equals(new ResourceLocation("tconstruct", "slime_helmet")) && !definition.getId().equals(new ResourceLocation("tconstruct", "plate_shield")))
+      .sorted(Comparator.comparingInt(a -> StationSlotLayoutLoader.getInstance().get(a.getId()).getSortIndex()))
+      .map(ToolRecipes::new)
+      .toList();
+    register.addRecipes(TConstructJEIConstants.TOOL_RECIPES, toolRecipes);
```

Comment:
This is wrong. Having a tool definition does not make it have a recipe. Just fetch the recipes.